Azure/sonic-buildimage
# [202012][device/celestica]: Add thermalctld support on Haliburton platform APIs

**- Why I did it**
- The thermalctld daemon on the Pmon docker requires support from the thermal manager API

**- How I did it**
- Cherry picked from : cfda77b3de1fc97f320213800d1523cd4809d6e5
> - Removed the old function for detecting a faulty fan.
> - Removed the old function for detecting excess temperature.
> - Implement thermal_manager APIs based on ThermalManagerBase
> - Implement thermal_conditions APIs based on ThermalPolicyConditionBase
> - Implement thermal_actions APIs based on ThermalPolicyActionBase
> - Implement thermal_info APIs based on ThermalPolicyInfoBase
> - Add thermal_policy.json

**- How to verify it**
- Check the fan speed during temperature changes.
- Examine events that will occur after the temperature has exceeded the threshold.
- Check for events that will occur after the fan is removed or the fan is not working properly.

**- Which release branch to backport (provide reason below if selected)**
